x-pack/filebeat/module/zeek/mqtt_publish/_meta/fields.yml

```diff
@@ -0,0 +1,41 @@
+- name: mqtt_publish
+  type: group
+  default_field: false
+  description: >
+    Fields exported by the Zeek mqtt_publish log.
+  fields:
+    - name: from_client
+      type: boolean
+      description: >
+        Indicates if the message was published by the client of this connection
+        or published to the client.
+
+    - name: retain
+      type: boolean
+      description: >
+        Indicates if the message was to be retained by the server.
+
+    - name: qos
+      type: keyword
+      description: >
+        QoS level set for the message.
+
+    - name: status
+      type: keyword
+      description: >
+        Status of the published message.
+
+    - name: topic
+      type: keyword
+      description: >
+        Topic the message was published to.
+
+    - name: payload
+      type: keyword
+      description: >
+        Payload of the message.
+
+    - name: payload_len
+      type: long
+      description: >
+        The actual length of the payload in the case the payload.
```

Comment:
Looking at MQTT::PublishInfo at https://docs.zeek.org/en/master/scripts/policy/protocols/mqtt/main.zeek.html.  It looks like there are fields like ack,rec,rel,comp & qos_level that could be present.  If they could show up in the logs we should add them.

Reply:
Yes, if are documented probably these fields can appear. so better to add. At that time when I tested, these fields didn't appear.